lib/vyasa/adapters/bind.ex

```diff
@@ -0,0 +1,41 @@
+defmodule Vyasa.Adapters.Binding do
+  @moduledoc """
+  Bindings that unite cross reference Archetypal Data Structs, they can be both persistent and virtual
+  """
+
+  use Ecto.Schema
+  import Ecto.Changeset
+
+  alias Vyasa.Written.{Verse, Source, Chapter}
+
+
+  embedded_schema do
+    belongs_to :chapter, Chapter, type: :integer, references: :no, foreign_key: :chapter_no
+    belongs_to :verse, Verse, foreign_key: :verse_id, type: :binary_id
+    belongs_to :source, Source, foreign_key: :source_id, type: :binary_id
+  end
+
+  def changeset(event, attrs) do
+    event
+    |> cast(attrs, [:verse_id, :voice_id, :source_id])
+  end
+
+  def cast(attrs) do
+    %__MODULE__{}
+    |> Vyasa.Repo.preload(__schema__(:associations))
+    |> cast(attrs, [])
+    |> put_all_assoc(attrs)
+    |> apply_action(:binded)
+  end
+
+  def put_all_assoc(changeset, attrs) do
```

Comment:
so yes, assocs are already loaded 

can't do a cast_assoc because the assocs are already casted. 

so we want an "update" clause for all the assocs. it's the Binding virtual assoc that is being "updated" 

^ that's the goal for the put_all_assoc()

Reply:
note persistence is done @ the context-level, never the changesetlevel